.github/workflows/nix_flake_update.yml

```diff
@@ -0,0 +1,26 @@
+---
+name: Nix Flake Update
+
+on:  # yamllint disable-line rule:truthy
+  workflow_dispatch:
+
+  schedule:
+    - cron: "0 0 1 * *"
+
+jobs:
+  nix-flake-update:
+    runs-on: ubuntu-22.04
+
+    permissions:
+      contents: write
+      pull-requests: write
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: DeterminateSystems/nix-installer-action@v16
+      - uses: DeterminateSystems/magic-nix-cache-action@v8
+
+      - uses: DeterminateSystems/update-flake-lock@v24
+        with:
+          pr-labels: dependencies
+          pr-title: "stylix: update all flake inputs"
```

Comment:
> We should use the GitHub App from #664 to allow GitHub Actions to run on the created pull request.
>
> -- https://github.com/danth/stylix/pull/520#pullrequestreview-2492953267

Upstream suggests adding an authentication token:

> ### With a Personal Authentication Token
>
> By providing a Personal Authentication Token, the PR will be submitted in a way that bypasses this limitation (GitHub will essentially think it is the owner of the PAT submitting the PR, and not an Action).
> You can create a token by visiting https://github.com/settings/tokens and select at least the `repo` scope. For the new fine-grained tokens, you need to enable read and write access for "Contents" and "Pull Requests" permissions. Then, store this token in your repository secrets (i.e. `https://github.com/<USER>/<REPO>/settings/secrets/actions`) as `GH_TOKEN_FOR_UPDATES` and set up your workflow file like the following:
>
> ```yaml
> name: update-flake-lock
> on:
>   workflow_dispatch: # allows manual triggering
>   schedule:
>     - cron: '0 0 * * 1,4' # Run twice a week
>
> jobs:
>   lockfile:
>     runs-on: ubuntu-latest
>     steps:
>       - name: Checkout repository
>         uses: actions/checkout@v4
>       - name: Install Nix
>         uses: DeterminateSystems/nix-installer-action@v1
>       - name: Update flake.lock
>         uses: DeterminateSystems/update-flake-lock@vX
>         with:
>           token: ${{ secrets.GH_TOKEN_FOR_UPDATES }}
> ```
>
> -- https://github.com/DeterminateSystems/update-flake-lock/blob/0ba11186640dde09ade3e5599aabb2a57f28d8aa/README.md?plain=1#L185-L209

I assume we have to add the following and properly setup `secrets.GH_TOKEN_UPDATE_FLAKE_LOCK`:

```suggestion
          pr-title: "stylix: update all flake inputs"
          token: ${{ secrets.GH_TOKEN_UPDATE_FLAKE_LOCK }}
```

Reply:
I believe the app token functions the same as a PAT, except that any actions taken by it use the identity of the app rather than a personal account.

Reply:
> I believe the app token functions the same as a PAT, except that any actions taken by it use the identity of the app rather than a personal account.

Which of the two approaches should we use?